### DISCUSSION TITLE
nx.from_pandas_adjacency does not persist all edges

### DISCUSSION THREAD

**Discussion by 13guff13** [networkx/networkx#6502] [Q&A] [answered]
```
import pandas as pd
import networkx as nx

df = pd.DataFrame([[5, 4], [2, 3]])
print(df)
G = nx.from_pandas_adjacency(df.T)
G.name = "Graph from pandas adjacency matrix"
print(G)

print(G[0][0])
print(G[0][1])
print(G[1][0])
print(G[1][1])
```
{'weight': 5}
{'weight': 4}
{'weight': 4}
{'weight': 3}

there is no edge with weight 2.

how to make it persist all edges ([5, 4], [2, 3])

**Comment by navyagarwal** [comment, marked as answer]
The method `nx.from_pandas_adjacency()` converts the dataframe to nx.Graph by default and nx.Graph does not allow parallel edges. Thus, there is a single edge (0,1) with a single weight value.

However, you can specify the type of graph you want to build by using the parameter `create_using`.

Here's the output you want to get -

<img width="425" alt="image" src="https://user-images.githubusercontent.com/82928853/224717814-6eff7f57-a644-430d-957e-5ef3eed61670.png">

**Reply by 13guff13** [reply]
I wasn't aware of this parameter, the docs doesn't mention this parameter. 
https://networkx.org/documentation/stable/reference/generated/networkx.convert_matrix.to_pandas_adjacency.html

with `create using` it working as expected, thank you :)

**Reply by navyagarwal** [reply]
@13guff13 No problem :) 

Although the docs do mention it here -> https://networkx.org/documentation/stable/reference/generated/networkx.convert_matrix.from_pandas_adjacency.html#networkx.convert_matrix.from_pandas_adjacenc
